src/backend/access/external/pxfheaders.c

```diff
@@ -65,14 +65,18 @@ void build_http_header(PxfInputData *input)
 	
 	if (proj_info != NULL && proj_info->pi_isVarList)
 	{
-		List* qualsAttributes = extractPxfAttributes(input->quals);
+		bool qualsAreSupported = true;
+		List* qualsAttributes = extractPxfAttributes(input->quals, &qualsAreSupported);
```

Comment:
qualsAreSupported seems redundant. Is there any scneario when qualsAttributes is not null when we don't want to support projection info ?

Reply:
qualsAttributes could be null when WHERE clause is empty, and in this case, we still want to use projection info. But if we don't support some operation in WHERE clause - we just report qualsAreSupported flag and projection info will not be used.